server/src/main/java/org/cloudfoundry/identity/uaa/zone/VmidentityIdentityZoneProvisioning.java

```diff
@@ -79,42 +79,84 @@ public IdentityZone retrieveBySubdomain(String subdomain) {
+        zone.setSubdomain(t.getName());
+        zone.setDescription(t._longName);
+        zone.setVersion(1);
+        
+        return zone;
+    }
+    
+    private void configureIdentityZone(String tenantName, IdentityZone zone) throws Exception {
+        IdentityZoneConfiguration config = this._configs.get(tenantName);
+        if (config == null) {
+            config = new IdentityZoneConfiguration();
+            this._configs.put(tenantName, config);
+        }
+        
+        Map<String, String> keys = new HashMap<String, String>();
+        PrivateKey privateKey = _idmClient.getTenantPrivateKey(tenantName);
+        StringWriter writer = new StringWriter();
+        PEMWriter pemWriter = new PEMWriter(writer);
+        pemWriter.writeObject(privateKey);
+        pemWriter.close();
+        
+        keys.put("key-id-1", writer.toString());
+        config.getTokenPolicy().setKeys(keys);
+        config.getTokenPolicy().setActiveKeyId("key-id-1");
+        config.getTokenPolicy().setAccessTokenValidity(
+                Ints.saturatedCast(_idmClient.getMaximumBearerTokenLifetime(tenantName)));
+        config.getTokenPolicy().setRefreshTokenValidity(
+                Ints.saturatedCast(_idmClient.getMaximumBearerRefreshTokenLifetime(tenantName)));
+        
+        zone.setConfig(config);
+    }
 
-    private IdentityZone identityZoneForTenant(Tenant t, boolean systemTenant) {
-        IdentityZone identityZone = null;
-        if (systemTenant) {
-            identityZone = IdentityZone.getUaa();
+    private IdentityZone getIdentityZoneFromId(String id) throws Exception {
+        String tenantName = id;
+        IdentityZone zone = null;
+        
+        if (id.equalsIgnoreCase(IdentityZone.getUaa().getId())) {
+            tenantName = this._idmClient.getSystemTenant();
+            zone = IdentityZone.getUaa();
         } else {
-            identityZone = new IdentityZone();
-
-            identityZone.setId(t.getName());
-            identityZone.setVersion(1);
-            // identityZone.setCreated(rs.getTimestamp(3));
-            // identityZone.setLastModified(rs.getTimestamp(4));
-            identityZone.setName(t.getName());
-            identityZone.setSubdomain(t.getName());
-            identityZone.setDescription(t._longName);
+            zone = identityZoneForTenant(tenantName);
         }
-
-        // todo: default for now, should implement
-        IdentityZoneConfiguration config = new IdentityZoneConfiguration();
-        if (this._configs.containsKey(t.getName())) {
-            config = this._configs.get(t.getName());
+        
+        configureIdentityZone(tenantName, zone);
+        
+        return zone;
+    }
+    
+    private IdentityZone getIdentityZoneFromTenant(String tenantName) throws Exception {
+        IdentityZone zone = null;
+        String systemTenant = this._idmClient.getSystemTenant();
+        
+        if (tenantName.equals(systemTenant)) {
```

Comment:
equalsIgnorecase... or you can add isSystemTenant util function to vmidentity utils.